libflux/src/flux/formatter/mod.rs

```diff
@@ -1,32 +1,57 @@
 #![allow(missing_docs)]
-use crate::ast::{self, walk::Node};
+use crate::ast::{self, walk::Node, File};
+use crate::parser::parse_string;
 use crate::Error;
 
 use chrono::SecondsFormat;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub fn format_from_js_file(js_file: JsValue) -> String {
+    if let Ok(file) = js_file.into_serde::<File>() {
+        if let Ok(converted) = convert_to_string(&file) {
+            return converted;
+        }
+    }
+    "".to_string()
+}
+
+pub fn convert_to_string(file: &File) -> Result<String, String> {
+    let mut formatter = Formatter::default();
+    formatter.format_file(file, true);
+    formatter.output()
+}
+
+pub fn format(contents: String) -> Result<String, String> {
+    let file = parse_string("", contents.as_str());
+
+    convert_to_string(&file)
+}
 
 pub struct Formatter {
     builder: String,
     indentation: u32,
     err: Option<Error>,
 }
 
-impl Formatter {
-    // giving too much capacity can significantly boost the performance
-    // see https://github.com/hoodie/concatenation_benchmarks-rs
-    pub fn new(cap: usize) -> Formatter {
+impl Default for Formatter {
+    fn default() -> Self {
         Formatter {
-            builder: String::with_capacity(cap),
+            builder: String::new(),
```

Comment:
the capacity is very helpful

Reply:
seems like a bad interface to require the user to have the length of the formatted text before they even run the formatter

Reply:
doubly so since we don't even require the original script but instead a File node.

Reply:
it's looks exactly like what you changed before. But with this capacity, we can saved thousand times of heap allocations.

Reply:
but you have to have that value when you create a formatter